src/aind_dynamic_foraging_basic_analysis/metrics/session_metrics.py

```diff
@@ -0,0 +1,217 @@
+"""
+
+    Consolidated session metric tool
+    dfs = nwb_utils.create_trials_df(your_nwb)
+
+"""
+
+import re
+
+import numpy as np
+import pandas as pd
+
+from aind_dynamic_foraging_basic_analysis import compute_foraging_efficiency
+
+# Copied metrics are from process_nwbs.py in bonsai basic
+# NOTE: finished_rate_with_autowater is the same as the calculated total response rate
+LEFT, RIGHT, IGNORE = 0, 1, 2
+
+
+def session_metrics(nwb):
+    """
+    Compute all session level metrics
+
+    Includes session level metadata as a temporary organizer
+    Includes majority of metrics from process_nwbs.py
+
+    New addition: chosen_probability - average difference between the chosen probability
+    and non-chosen probability / the difference between the largest and smallest probability
+    in the session
+    """
+
+    if not hasattr(nwb, "df_trials"):
+        print("You need to compute dfs: nwb_utils.create_trials_df(nwb)")
+        return
+
+    df = nwb.df_trials.copy()
+
+    # METADATA PLACEHOLDER
+    session_start_time = nwb.session_start_time
+    session_date = session_start_time.strftime("%Y-%m-%d")
+    subject_id_from_meta = nwb.subject.subject_id
+
+    # Parse the file name for suffix
+    old_re = re.match(
+        r"(?P<subject_id>\d+)_(?P<date>\d{4}-\d{2}-\d{2})(?:_(?P<n>\d+))?\.json", nwb.session_id
+    )
+
+    if old_re is not None:
+        subject_id, session_date, nwb_suffix = old_re.groups()
+        nwb_suffix = int(nwb_suffix) if nwb_suffix is not None else 0
+    else:
+        subject_id, session_date, session_json_time = re.match(
+            r"(?P<subject_id>\d+)_(?P<date>\d{4}-\d{2}-\d{2})(?:_(?P<time>.*))\.json",
+            nwb.session_id,
+        ).groups()
+        nwb_suffix = int(session_json_time.replace("-", ""))
+
+    # Verify metadata matches
```

Comment:
We don't need these assert statements, since that is now handled by the pipeline that makes the NWB files